src/QuadForm/Quad/ZLattices.jl

```diff
@@ -491,16 +493,52 @@ function _root_lattice_A(n::Int)
   return z
 end
 
+function _root_lattice_D(n::Int)
+  n < 2 && error("Parameter ($n) for root lattices of type :D must be greater of equal to 2")
+  if n == 2
+    G = matrix(ZZ, [2 0 ;0 2])
+  elseif n == 3
+    return _root_lattice_A(n)
+  else
+    G = zero_matrix(ZZ, n, n)
+    G[1,3] = G[3,1] = -1
+    for i in 1:n
+      G[i,i] = 2
+      if i in 2:n-1
```

Comment:
```suggestion
      if 2 <= i <= n-1
```

Reply:
Now that we are moving equality of lattices, should I keep the definition of `D3` like this, being equal to `A3` or should I make standard like `D4`,`D5`, and so on ? In this case `D3` and `A3` will be isometric but not equal.

Reply:
It doesn't really matter I think. So you can keep it. There is also an elegant description of the `A,D,E` root lattices as a sublattice of the standard euclidean space of dimension `n+1`. But lets keep the ambient keyword for later :-).